File: digital-library-frontend/src/services/bookService.js
import axiosInstance from '../api/axiosConfig';

const bookService = {
  getAllBooks: () => axiosInstance.get('/api/books'),
  getBookById: (id) => axiosInstance.get(`/api/books/${id}`),
  searchBooks: (keyword) => axiosInstance.get(`/api/books/search?keyword=${keyword}`),
  getByCategory: (category) => axiosInstance.get(`/api/books/category/${category}`),

  // Admin
  addBook: (data) => axiosInstance.post('/api/admin/books', data),
  updateBook: (id, data) => axiosInstance.put(`/api/admin/books/${id}`, data),
  deleteBook: (id) => axiosInstance.delete(`/api/admin/books/${id}`),
};

export default bookService;

File: digital-library-backend/src/main/java/com/digitallibrary/service/BookService.java
package com.digitallibrary.service;

import com.digitallibrary.dto.BookRequest;
import com.digitallibrary.dto.BookResponse;
import java.util.List;

public interface BookService {
    List<BookResponse> getAllBooks();
    BookResponse getBookById(Long id);
    List<BookResponse> searchBooks(String keyword);
    List<BookResponse> getBooksByCategory(String category);
    BookResponse addBook(BookRequest request);
    BookResponse updateBook(Long id, BookRequest request);
    void deleteBook(Long id);
}
